From fcf08ed99d1421158ca913a7b6a4893c31ee1bd5 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Tue, 15 Dec 2020 17:02:17 +0000
Subject: [PATCH] overlays: mpu6050: Add 'addr' parameter

The mpu6050 starts up at address 0x68 by default, but can be set to
0x69 if the ADO pin is pulled high. Give the overlay an addr parameter
to allow devices at the alternate address to be used.

See: https://github.com/Hexxeh/rpi-firmware/issues/252

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/README              | 1 +
 arch/arm/boot/dts/overlays/mpu6050-overlay.dts | 1 +
 2 files changed, 2 insertions(+)

--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -1876,6 +1876,7 @@ Name:   mpu6050
 Info:   Overlay for i2c connected mpu6050 imu
 Load:   dtoverlay=mpu6050,<param>=<val>
 Params: interrupt               GPIO pin for interrupt (default 4)
+        addr                    I2C address of the device (default 0x68)
 
 
 Name:   mz61581
--- a/arch/arm/boot/dts/overlays/mpu6050-overlay.dts
+++ b/arch/arm/boot/dts/overlays/mpu6050-overlay.dts
@@ -24,5 +24,6 @@
 
         __overrides__ {
                 interrupt = <&mpu6050>,"interrupts:0";
+                addr = <&mpu6050>,"reg:0";
         };
 };
